charts/linkerd2/templates/prometheus.yaml

```diff
@@ -50,6 +50,13 @@ data:
 
       kubernetes_sd_configs:
       - role: node
+      metric_relabel_configs:
+      - source_labels: [__name__]
+        regex: '(container|machine)_(cpu|memory|network|fs)_(.+)'
+        action: keep
+      - source_labels: [__name__]
+        regex: 'container_memory_failures_total' # unneeded large metric
+        action: drop
```

Comment:
what do you think about limiting the metrics strictly to what we show in the dashboard? /cc @grampelberg 

```bash
$ curl -s "http://localhost:9090/api/v1/query?query=count(%7Bjob%3D%22kubernetes-nodes-cadvisor%22%7D)%20by(__name__)" |
  jq -r '.data.result[] | "\(.value[1]):\(.metric.__name__)" ' |
  sort -r -n |
  xargs -I {} sh -c '
    echo {} |
    cut -d ":" -f2 |
    grep -q -f /dev/stdin grafana/dashboards/kubernetes.json; echo $?: {}
  '
0: 62:container_memory_working_set_bytes
1: 62:container_memory_usage_bytes
1: 62:container_memory_swap
1: 62:container_memory_rss
1: 62:container_memory_max_usage_bytes
1: 62:container_memory_mapped_file
1: 62:container_memory_failcnt
1: 62:container_memory_cache
1: 62:container_cpu_user_seconds_total
0: 62:container_cpu_usage_seconds_total
1: 62:container_cpu_system_seconds_total
1: 62:container_cpu_load_average_10s
1: 30:container_fs_writes_total
1: 30:container_fs_reads_total
1: 27:container_fs_writes_bytes_total
1: 27:container_fs_reads_bytes_total
1: 17:container_network_transmit_packets_total
1: 17:container_network_transmit_packets_dropped_total
1: 17:container_network_transmit_errors_total
0: 17:container_network_transmit_bytes_total
1: 17:container_network_receive_packets_total
1: 17:container_network_receive_packets_dropped_total
1: 17:container_network_receive_errors_total
0: 17:container_network_receive_bytes_total
1: 4:container_fs_writes_merged_total
1: 4:container_fs_write_seconds_total
0: 4:container_fs_usage_bytes
1: 4:container_fs_sector_writes_total
1: 4:container_fs_sector_reads_total
1: 4:container_fs_reads_merged_total
1: 4:container_fs_read_seconds_total
0: 4:container_fs_limit_bytes
1: 4:container_fs_io_time_weighted_seconds_total
1: 4:container_fs_io_time_seconds_total
1: 4:container_fs_io_current
1: 4:container_fs_inodes_total
1: 4:container_fs_inodes_free
1: 2:container_cpu_cfs_throttled_seconds_total
1: 2:container_cpu_cfs_throttled_periods_total
1: 2:container_cpu_cfs_periods_total
1: 1:up
1: 1:scrape_series_added
1: 1:scrape_samples_scraped
1: 1:scrape_samples_post_metric_relabeling
1: 1:scrape_duration_seconds
0: 1:machine_memory_bytes
0: 1:machine_cpu_cores
```

(un)related: i just noticed that we're collecting `container_memory_usage_bytes` but grafana references `container_memory_usage`, so that dashboard may need an update (https://grafana.com/grafana/dashboards/315 is 2 years old). this is probably for another PR, but noting it here in case we do remove every metric above that is not referenced in the dashboard.

Reply:
Yup, I like the idea of keeping only those metrics that the dashboard needs; just wasn't sure if someone will ever want to query for those other cadvisor-related metrics, say in a federated environment, for alerts etc.